### PR TITLE
Lock JLine to previous version (3.23.0)

### PR DESCRIPTION
**Pull request by translatenix**
JLine 3.25.1 does not work with native image.

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -20,7 +20,9 @@ javaPoet = "1.+"
 javaxInject = "1"
 jansi = "2.+"
 jimfs = "1.+"
-jline = "3.+"
+# 3.25.1 doesn't work with native image
+# https://github.com/jline/jline3/pull/959
```
Comment (bioball) [inline gradle/libs.versions.toml:24]: I believe the issue is a new flag that they added to their `native-image.properties`

Our build error:

```
Error: Could not find option 'UnlockExperimentalVMOptions' from 'jar:file:///root/.gradle/caches/modules-2/files-2.1/org.jline/jline-native/3.25.1/7f58e474f7d94db5bf87b1fddf4fa646475779f2/jline-native-3.25.1.jar!/META-INF/native-image/org.jline/jline-native/native-image.properties'. Use -H:PrintFlags= to list all available options.
Could not find option 'UnlockExperimentalVMOptions' from 'jar:file:///root/.gradle/caches/modules-2/files-2.1/org.jline/jline-terminal/3.25.1/44e0e53397c39d01c525bad3735097596a7d889a/jline-terminal-3.25.1.jar!/META-INF/native-image/org.jline/jline-terminal/native-image.properties'. Use -H:PrintFlags= to list all available options.
Could not find option 'UnlockExperimentalVMOptions' from 'jar:file:///root/.gradle/caches/modules-2/files-2.1/org.jline/jline-terminal-jansi/3.25.1/ee87b1238f01fe352ff4b78656e3dd4039bee727/jline-terminal-jansi-3.25.1.jar!/META-INF/native-image/org.jline/jline-terminal-jansi/native-image.properties'. Use -H:PrintFlags= to list all available options.
```

**Review by bioball — APPROVED** [review]

File: gradle/libs.versions.toml (hunk 1 above)
Comment (bioball) [inline gradle/libs.versions.toml:24]:
```suggestion
```

Comment (translatenix) [reply gradle/libs.versions.toml]: Sounds like yet another issue. When I tried building Pkl with GraalVM 23 and JLine 3.25.1 a while ago, I ran into the typo issue. AFAICT, 3.25.1 is totally broken wrt. native image.

Comment (translatenix) [reply gradle/libs.versions.toml]: This fix is definitely required. It may not be sufficient.

**Pull request merged**